src/clojure/nrepl/core.clj

```diff
@@ -64,6 +64,14 @@
     (concat head (take 1 tail))))
 
 (defn- delimited-transport-seq
+  "Sends a message via client and returns the head of client's response seq
```

Comment:
Seems to me this returns a function that will do all of the above. The actual `delimited-transport-seq` doesn't do any sending. It's also not returning a seq, which makes its name slightly confusing, but that's not a big deal I guess (and I don't have an idea for a better name).